storage/invoke.go

```diff
@@ -84,7 +84,21 @@ func setInvocationHeaders(ctx context.Context, invocationID string, attempts int
 	invocationHeader := fmt.Sprintf("gccl-invocation-id/%v gccl-attempt-count/%v", invocationID, attempts)
 	xGoogHeader := strings.Join([]string{invocationHeader, xGoogDefaultHeader}, " ")
 
-	ctx = callctx.SetHeaders(ctx, xGoogHeaderKey, xGoogHeader)
+	// TODO: remove this once the respective transport packages merge xGoogHeader.
+	// Also remove gl-go at that time, as it will be repeated.
+	hdrs := callctx.HeadersFromContext(ctx)
+	for _, v := range hdrs[xGoogHeaderKey] {
+		xGoogHeader = strings.Join([]string{xGoogHeader, v}, " ")
+	}
+
+	if hdrs[xGoogHeaderKey] != nil {
+		// Replace the key instead of adding it, if there was anything to merge with.
+		hdrs[xGoogHeaderKey] = []string{xGoogHeader}
+	} else {
+		// TODO: keep this line when removing the above code.
+		ctx = callctx.SetHeaders(ctx, xGoogHeaderKey, xGoogHeader)
+	}
```

Comment:
This looks right but could you add a test? Possibly could be added as a case in the existing tests for invoke, but if a separate one is easier that's fine too.

Reply:
Added. I think a separate one is better because it should be removed with the code.